src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value,null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. It seems you're doing this correctly, but make sure not to introduce getters and setters for the Node class, as they are not needed.